src/bt.rs

```diff
@@ -570,6 +570,74 @@ where
+            ble_ll_jitter_usecs: crate::sys::BLE_LL_JITTER_USECS_N as _,
+            ble_ll_sched_max_adv_pdu_usecs: crate::sys::BLE_LL_SCHED_MAX_ADV_PDU_USECS_N as _,
+            ble_ll_sched_direct_adv_max_usecs: crate::sys::BLE_LL_SCHED_DIRECT_ADV_MAX_USECS_N as _,
+            ble_ll_sched_adv_max_usecs: crate::sys::BLE_LL_SCHED_ADV_MAX_USECS_N as _,
+            ble_scan_rsp_data_max_len: crate::sys::DEFAULT_BT_LE_SCAN_RSP_DATA_MAX_LEN_N as _,
+            ble_ll_cfg_num_hci_cmd_pkts: crate::sys::BLE_LL_CFG_NUM_HCI_CMD_PKTS_N as _,
+            ble_ll_ctrl_proc_timeout_ms: crate::sys::BLE_LL_CTRL_PROC_TIMEOUT_MS_N,
+            nimble_max_connections: crate::sys::DEFAULT_BT_LE_MAX_CONNECTIONS as _,
+            ble_whitelist_size: crate::sys::DEFAULT_BT_NIMBLE_WHITELIST_SIZE as _,
+            ble_acl_buf_size: crate::sys::DEFAULT_BT_LE_ACL_BUF_SIZE as _,
+            ble_acl_buf_count: crate::sys::DEFAULT_BT_LE_ACL_BUF_COUNT as _,
+            ble_hci_evt_buf_size: crate::sys::DEFAULT_BT_LE_HCI_EVT_BUF_SIZE as _,
+            ble_multi_adv_instances: crate::sys::DEFAULT_BT_LE_MAX_EXT_ADV_INSTANCES as _,
+            ble_ext_adv_max_size: crate::sys::DEFAULT_BT_LE_EXT_ADV_MAX_SIZE as _,
+            controller_task_stack_size: crate::sys::NIMBLE_LL_STACK_SIZE as _,
+            controller_task_prio: crate::sys::ESP_TASK_BT_CONTROLLER_PRIO as _,
+            controller_run_cpu: 0,
+            enable_qa_test: crate::sys::RUN_QA_TEST as _,
+            enable_bqb_test: crate::sys::RUN_BQB_TEST as _,
+            enable_uart_hci: crate::sys::HCI_UART_EN as _,
+            ble_hci_uart_port: crate::sys::DEFAULT_BT_LE_HCI_UART_PORT as _,
+            ble_hci_uart_baud: crate::sys::DEFAULT_BT_LE_HCI_UART_BAUD,
+            ble_hci_uart_data_bits: crate::sys::DEFAULT_BT_LE_HCI_UART_DATA_BITS as _,
+            ble_hci_uart_stop_bits: crate::sys::DEFAULT_BT_LE_HCI_UART_STOP_BITS as _,
+            ble_hci_uart_flow_ctrl: crate::sys::DEFAULT_BT_LE_HCI_UART_FLOW_CTRL as _,
+            ble_hci_uart_uart_parity: crate::sys::DEFAULT_BT_LE_HCI_UART_PARITY as _,
+            enable_tx_cca: crate::sys::DEFAULT_BT_LE_TX_CCA_ENABLED as _,
+            cca_rssi_thresh: (256 - crate::sys::DEFAULT_BT_LE_CCA_RSSI_THRESH) as _,
+            sleep_en: crate::sys::NIMBLE_SLEEP_ENABLE as _,
+            coex_phy_coded_tx_rx_time_limit: crate::sys::DEFAULT_BT_LE_COEX_PHY_CODED_TX_RX_TLIM_EFF
+                as _,
+            dis_scan_backoff: crate::sys::NIMBLE_DISABLE_SCAN_BACKOFF as _,
+            ble_scan_classify_filter_enable: 1,
+            main_xtal_freq: crate::sys::CONFIG_XTAL_FREQ as _,
+            #[cfg(esp32c2)]
+            version_num: unsafe { crate::sys::esp_ble_get_chip_rev_version() },
+            #[cfg(esp32c6)]
+            version_num: unsafe { crate::sys::efuse_hal_chip_revision() },
+            #[cfg(not(esp32c2))]
+            cpu_freq_mhz: crate::sys::CONFIG_ESP_DEFAULT_CPU_FREQ_MHZ as _,
```

Comment:
can c2 work correctly with a default cpu_freq_mhz here? Or do we need to find something that exist on c2 that represent the equivalent of DEFAULT_CPU_FREQ_MHZ?

Reply:
The BT controller for esp32c2 does not even have this field.